internal/logging/sentry.go

```diff
@@ -0,0 +1,24 @@
+package logging
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/RHEnVision/provisioning-backend/internal/config"
+	"github.com/RHEnVision/provisioning-backend/internal/logging/sentry_writer"
+	"github.com/getsentry/sentry-go"
+)
+
+// sentryWriter creates a zerolog writer for sentry.
+// Uses github.com/archdx/zerolog-sentry which is very simple wrapper.
+func sentryWriter(dsn string) (io.Writer, func(), error) {
+	wr, err := sentry_writer.New(dsn)
+	if err != nil {
+		return nil, func() {}, fmt.Errorf("cannot initialize sentry: %w", err)
+	}
+	sentry.ConfigureScope(func(scope *sentry.Scope) {
+		scope.SetTag("stream", config.BinaryName())
+	})
+	// close function returns error, but it's always nil, so we are ignoring it here
+	return wr, func() { _ = wr.Close() }, nil
+}
```

Comment:
Not a huge fan of this approach which serializes the events into JSON and then parsing them back into memory. But as long as this is only done for error/fatal events, I am fine. I will be looking for a proof this is not done for info and lower-level events tho.

I was hoping we could use zerolog's hook system instead.

Reply:
So it looks like there is no other way for zerolog :(